mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/checkertaskfilterdialog/CheckerTaskFilterDialogFragment.kt

```diff
@@ -0,0 +1,176 @@
+package com.mifos.mifosxdroid.dialogfragments.checkertaskfilterdialog
+
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import android.content.Context
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.fragment.app.DialogFragment
+import com.mifos.mifosxdroid.R
+import com.mifos.mifosxdroid.core.MifosBaseActivity
+import com.mifos.mifosxdroid.online.checkerinbox.CheckerInboxViewModel
+import com.mifos.mifosxdroid.online.checkerinbox.CheckerInboxViewModelFactory
+import com.mifos.mifosxdroid.uihelpers.MFDatePicker
+import com.mifos.utils.FragmentConstants
+import kotlinx.android.synthetic.main.dialog_fragment_checker_task_filter.*
+import java.lang.ClassCastException
+import java.sql.Timestamp
+import java.text.SimpleDateFormat
+import javax.inject.Inject
+
+class CheckerTaskFilterDialogFragment : DialogFragment(), MFDatePicker.OnDatePickListener,
+        AdapterView.OnItemSelectedListener {
+
+    private lateinit var mOnInputSelected: OnInputSelected
+
+    private lateinit var datePickerFromDate: DialogFragment
+    private lateinit var datePickerToDate: DialogFragment
+    private lateinit var mCurrentDateView: View
+    private lateinit var selectedAction: String
+    private lateinit var selectedEntity: String
+
+    private lateinit var actionOptionsList: MutableList<String>
+    private lateinit var entityOptionsList: MutableList<String>
+    private lateinit var actionOptionsAdapter: ArrayAdapter<String>
+    private lateinit var entityOptionsAdapter: ArrayAdapter<String>
+
+    private lateinit var fromDate: String
+    private lateinit var toDate: String
+
+    @Inject
+    lateinit var factory: CheckerInboxViewModelFactory
+    private lateinit var viewModel: CheckerInboxViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        actionOptionsList = mutableListOf()
+        entityOptionsList = mutableListOf()
+        actionOptionsList.add("ALL")
```

Comment:
Make some constant(s) for these types. It'll make it convenient for developers to makes changes later.